src/main/java/com/google/devtools/build/lib/actions/SpawnResult.java

```diff
@@ -358,6 +363,15 @@ public String getDetailMessage(
       }
       return messagePrefix + " failed" + reason + explanation;
     }
+
+    @Nullable
+    @Override
+    public InputStream getInMemoryOutput(ActionInput output) {
+      if (inMemoryOutputFile != null && inMemoryOutputFile.equals(output)) {
```

Comment:
Why do we need this check? I see that we want to check that the file that we have in-memory and the file that we are requesting are the same, and only after that we return its content. But do you see any potential use cases issues when this check is needed?

Reply:
The contract of `getInMemoryOutput(ActionInput)` requires us to have the check, as a caller can pass in an arbitrary `ActionInput`. You are correct that currently there are only two callers of this method that only ever pass in a specific `ActionInput`. However, this might change...